video/decode/vd_lavc.c

```diff
@@ -568,6 +572,19 @@ static void update_image_params(struct dec_video *vd, AVFrame *frame,
     vd_ffmpeg_ctx *ctx = vd->priv;
     struct MPOpts *opts = ctx->opts;
 
+#if HAVE_AVUTIL_MASTERING_METADATA
+    // Get the reference peak (for HDR) if available. This is cached into ctx
+    // when it's found, since it's not available on every frame (and seems to
+    // be only available for keyframes)
+    AVFrameSideData *sd = av_frame_get_side_data(frame,
+                          AV_FRAME_DATA_MASTERING_DISPLAY_METADATA);
+    if (sd) {
+        AVMasteringDisplayMetadata *mdm = (AVMasteringDisplayMetadata *)sd->data;
+        if (mdm->has_luminance)
+            ctx->cached_hdr_peak = av_q2d(mdm->max_luminance);
```

Comment:
There's `av_stream_get_side_data()`. And in case it causes confusion: this side data can be both on AVStream and AVPacket. If it's on AVStream it's supposed to be constant over the whole stream (but on the demuxer level - the decoder can still output stuff from the hevc bitstream). The AVPacket side data is actually passed along to the decoder, which might just output a copy of it. In theory, we could maybe inject the AVStream side data as AVPacket side data for every stream to get it to the decoder automatically. Not sure what's best.